threescale/api/utilities.go

```diff
@@ -36,6 +37,30 @@ func (m Metrics) ComputeAffectedMetrics(hierarchy Hierarchy) Metrics {
 	return metrics
 }
 
+// SubtractHierarchyFromMetrics takes the provided hierarchy structure, and uses it
+// to determine how the metrics, m, are affected, decrementing parent metrics
+// based on the value of the parents child/children metrics.
+// Returns new Metrics, leaving metrics m in it's original state.
+func (m Metrics) SubtractHierarchyFromMetrics(hierarchy Hierarchy) Metrics {
+	metrics := m.DeepCopy()
+
+	for parent, children := range hierarchy {
+		for metric, v := range metrics {
+			if contains(metric, children) {
+				if value, known := metrics[parent]; known {
```

Comment:
this makes me think in the code right above for incrementing (not changed in this PR except for adding a comment), perhaps we can do the addition (since we lookup the value anyways) and just call `metrics.Set()`?